beacon-chain/verification/data_column.go

```diff
@@ -257,17 +257,26 @@ func (dv *RODataColumnsVerifier) ValidProposerSignature(ctx context.Context) (er
 			continue
 		}
 
-		columnVerificationProposerSignatureCache.WithLabelValues("miss").Inc()
+		// Ensure the expensive signature verification is only performed once for
+		// concurrent requests for the same signature data.
+		if _, err, _ = dv.sg.Do(signatureData.string(), func() (any, error) {
```

Comment:
Is there a way to test this works as it is? I guess its hard and the only way to verify this is running a node and measure live data?

Reply:
Both `seen` and non `seen` cases are tested.
Testing the single flight feature itself is not done, because it could be racy in tests.
However, we can trust the golang stdlib to work as expected.
(And yes it tested in live.)